host_core/native/hostcore_wasmcloud_native/src/oci.rs

```diff
@@ -50,32 +53,57 @@ pub(crate) async fn fetch_oci_path(
             "Fetching images tagged 'latest' is currently prohibited in this host. This option can be overridden with WASMCLOUD_OCI_ALLOW_LATEST".into());
     }
     let cf = cached_file(img).await?;
-    if tokio::fs::metadata(&cf).await.is_err() {
-        let auth = determine_auth(img, creds_override);
-        let img = oci_distribution::Reference::from_str(img)?;
 
-        let protocol =
-            oci_distribution::client::ClientProtocol::HttpsExcept(allowed_insecure.to_vec());
-        let config = oci_distribution::client::ClientConfig {
-            protocol,
-            ..Default::default()
-        };
-        let mut c = oci_distribution::Client::new(config);
-        let imgdata = pull(&mut c, &img, &auth).await;
+    let auth = determine_auth(img, creds_override);
+    let img = oci_distribution::Reference::from_str(img)?;
+
+    let protocol =
+        oci_distribution::client::ClientProtocol::HttpsExcept(allowed_insecure.to_vec());
+    let config = oci_distribution::client::ClientConfig {
+        protocol,
+        ..Default::default()
+    };
+    let mut c = oci_distribution::Client::new(config);
 
+    if tokio::fs::metadata(&cf).await.is_err() {     
+        let imgdata = pull(&mut c, &img, &auth).await;
         match imgdata {
             Ok(imgdata) => {
-                let mut f = tokio::fs::File::create(&cf).await?;
-                let content = imgdata
-                    .layers
-                    .into_iter()
-                    .flat_map(|l| l.data)
-                    .collect::<Vec<_>>();
-                f.write_all(&content).await?;
-                f.flush().await?;
+                cache_oci_image(&imgdata, &cf).await;
             }
             Err(e) => return Err(format!("Failed to fetch OCI bytes: {}", e).into()),
         }
+    }else{
+        let manifest = c.pull_manifest(&img, &auth).await;
+        match manifest{
+            Ok(manifest)=>{
+                let (oci_manifest, _) = manifest;
+                if let OciManifest::Image(image_manifest) = oci_manifest {
+                    
+                    let layers = image_manifest.layers;
+                    // The digest will be string like sha256:asdf, and the first 7 characters are removed
+                    let oci_sha = &layers[0].digest[7..];
```

Comment:
Two small notes here: I'd check that the `layers` vec is not empty so we don't get a panic here. Second is a nit that I think it would read clearer if you did `layers[0].digest.trim_left_matches("sha256:")`